klarna-order-management-for-woocommerce.php

```diff
@@ -61,6 +61,11 @@ public static function get_instance() {
 				self::$instance = new self();
 			}
 
+			// If called through a snippet, the init() call may not have been invoked yet, rendering certain properties null (such as the settings).
+			if ( ! isset( self::$instance->settings ) ) {
+				self::$instance->init();
+			}
+
```

Comment:
The `init` function runs on `plugins_loaded` which is very early in the WordPress actions [order](https://wp-kama.com/hooks/actions-order). Does that not run before the snippets are loaded in this case? Is it an issue with a snippet being ran instantly perhaps?

I can potentially see this causing other issues, for example if people try to run this before even WooCommerce is loaded. Would a better solution not be to say you need to run this method after x action instead? Similar to what WooCommerce does here: https://github.com/woocommerce/woocommerce/blob/9297409c5a705d1cd0ae65ec9b058271bd90851e/plugins/woocommerce/includes/wc-notice-functions.php#L25

Reply:
I found that it is being executed too early, while `plugins_loaded` has not yet been triggered. I agree with your suggested solution.